PxWeb/Code/Api2/DataSelection/SelectionHandler.cs

```diff
@@ -0,0 +1,303 @@
+                {
+                    // Content - Take all values
+                    selection.ValueCodes.AddRange(GetAllCodes(variable));
+                }
+                else if (variable.IsTime)
+                {
+                    // Time - Take the last value
+                    selection.ValueCodes.AddRange(GetTimeCodes(variable, 1));
+                }
+                else if (!variable.Elimination)
+                {
+                    // Take all values for the third variable
+                    selection.ValueCodes.AddRange(GetAllCodes(variable));
+                }
+                else
+                {
+                    // Select nothing for the other variables
+                }
+
+                selections.Add(selection);
+            }
+
+            return selections.ToArray();
+        }
+
+        /// <summary>
+        /// Returns default selection for all other cases
+        /// </summary>
+        /// <param name="model"></param>
+        /// <returns></returns>
+        private Selection[] GetDefaultSelectionAllOtherCases(PXModel model)
+        {
+            var selections = new List<Selection>();
+            int variableNumber = 1;
+
+            foreach (var variable in model.Meta.Variables)
+            {
+                var selection = new Selection(variable.Code);
+
+                if (variable.IsContentVariable)
+                {
+                    // Content - Take the first value
+                    selection.ValueCodes.AddRange(GetCodes(variable, 1));
+                }
+                else if (variable.IsTime)
+                {
+                    // Time - Take the last value
+                    selection.ValueCodes.AddRange(GetTimeCodes(variable, 1));
+                }
+                else if (!variable.Elimination && (variableNumber == 1 || variableNumber == 2))
+                {
+                    // Take all values for variable 1 and 2
+                    selection.ValueCodes.AddRange(GetAllCodes(variable));
+                    variableNumber++;
+                }
+                else if (!variable.Elimination && (variableNumber > 2))
+                {
+                    // Take 1 value 
+                    selection.ValueCodes.AddRange(GetCodes(variable, 1));
+                    variableNumber++;
+                }
+                else
+                {
+                    // Select nothing for the other variables
+                }
+
+                selections.Add(selection);
+            }
+
+            return selections.ToArray();
+        }
+
+        private string[] GetCodes(Variable variable, int count)
+        {
+            var codes = variable.Values.Take(count).Select(value => value.Code).ToArray();
+
+            return codes;
+        }
+
+        private string[] GetAllCodes(Variable variable)
+        {
+            var codes = variable.Values.Take(variable.Values.Count).Select(value => value.Code).ToArray();
```

Comment:
You could remove the .Take(variable.Values.Count) since you will take all codes.

Reply:
Removed code as suggested